cmd/auth/login.go

```diff
@@ -20,7 +37,33 @@ var loginCmd = &cobra.Command{
 		defer perisistentAuth.Close()
 		ctx, cancel := context.WithTimeout(cmd.Context(), loginTimeout)
 		defer cancel()
-		return perisistentAuth.Challenge(ctx)
+
+		var profileName string
+		profileFlag := cmd.Flag("profile")
+		if profileFlag != nil && profileFlag.Value.String() != "" {
+			profileName = profileFlag.Value.String()
+		} else {
+			profiles, err := getProfileNames()
+			if err != nil {
+				return err
+			}
+			profile, err := cmdio.SelectWithAdd(ctx, profiles, "~/.databrickscfg profile", "Add a new profile")
+			if err != nil {
+				return err
+			}
+			profileName = profile
```

Comment:
I was thinking a freeform text prompt with a suggested default.

For example, if I'm logging into `foobar.cloud.databricks.com` it would be nice to get a prompt for a profile name with the default of `foobar` filled in. By selecting from a list it's not possible to provide a sane default.

Reply:
With SelectWithAdd you can still do the free form right? (see the example of run in PR description). You can either choose from existing one or add a new one like in the free form